test/Tests/src/Data/Time/Date_Time_Spec.enso

```diff
@@ -867,6 +880,18 @@ foreign js js_array_datetimeCreate year month day hour minute second nanosecond
     }
     return [ new Date(year, month - 1, day, hour, minute, second, nanosecond / 1000000) ];
 
+foreign python python_datetime_impl year month day hour minute second nanosecond zone = """
+    import datetime
+
+    class Zone(datetime.tzinfo):
+        def __init__(self,zone):
+            self.zone = zone
+
+        def utcoffset(self, dt):
+            return datetime.timedelta(seconds=zone)
+
+    return datetime.datetime(year, month, day, hour, minute, second, int(nanosecond / 1000000), Zone(zone))
```

Comment:
I tried:
```diff
 python_datetime year month=1 day=1 hour=0 minute=0 second=0 nanosecond=0 zone=Time_Zone.system =
-    rules = Polyglot.invoke (ZoneId.of (zone.zone_id)) "getRules" []
-    zone_offset = Polyglot.invoke rules "getOffset" [ LocalDateTime.now ]
-    delta = Polyglot.invoke zone_offset "getTotalSeconds" []
-    Panic.catch Any (python_datetime_impl year month day hour minute second nanosecond delta) (err -> Error.throw (Time_Error.Error err.payload))
+    Panic.catch Any (python_datetime_impl year month day hour minute second nanosecond (Python_Zone.Value zone)) (err -> Error.throw (Time_Error.Error err.payload))
 
 python_parse text format="" =
     d = Date_Time.parse text format
@@ -880,6 +874,18 @@ foreign js js_array_datetimeCreate year month day hour minute second nanosecond
     }
     return [ new Date(year, month - 1, day, hour, minute, second, nanosecond / 1000000) ];
 
+type Python_Zone
+    Value zone:Time_Zone
+
+    total_seconds self dt =
+        rules = Polyglot.invoke (ZoneId.of (self.zone.zone_id)) "getRules" []
+        zone_offset = Polyglot.invoke rules "getOffset" [ dt ]
+        delta = Polyglot.invoke zone_offset "getTotalSeconds" []
+
+    zone_id self = self.zone.zone_id
+
+    dst self dt = Panic.throw "dst"
+
 foreign python python_datetime_impl year month day hour minute second nanosecond zone = """
     import datetime
 
@@ -888,7 +894,13 @@ foreign python python_datetime_impl year month day hour minute second nanosecond
             self.zone = zone
 
         def utcoffset(self, dt):
-            return datetime.timedelta(seconds=zone)
+            return datetime.timedelta(seconds=zone.total_seconds(dt))
+
+        def dst(self, dt):
+            return datetime.timedelta(seconds=zone.dst(dt))
+
+        def tzname(self, dt):
+            return zone.zone_id()
 
     return datetime.datetime(year, month, day, hour, minute, second, int(nanosecond / 1000000), Zone(zone))
```
but I am getting `StackOverflowError` as the `dt` is _"used again"_ to compute the `utcoffset`.

Reply:
What do you mean it is 'used again' to compute utcoffset?

This looks like a good direction, we just need to make sure it doesn't loop. I imagine we may need to do the conversion from Python to Java in the `total_seconds` helper in some more manual manner to avoid this endless loop? Maybe we should perform the conversion removing the timezone to get a `LocalDateTime` and then re-add the timezone afterwards? I'm not sure but there certainly must be a way to do it.

Reply:
This `Zone` approach is just a workaround to _fool the test_. No need to make it perfect. It shows what are the problems, but the real fix has to happen elsewhere: #7418